external-docs/docs/guides/csv-to-csvw-github-action.md

```diff
@@ -0,0 +1,73 @@
+
+* The CSV files and their configuration JSONs need to be inside the `csv/` folder.
+  
+* If a CSV file has a configuration JSON, the filename of the configuration JSON needs to match the filename of the CSV file. For example, `my_data.CSV` and `my_data.csv.json`.
+
+## User Manual
+
+CSV-to-CSV-W GitHub action can be used easily through the GitHub web console. Users who are experienced with GitHub can also clone the repository to the local machine and then commit files using the [GitHub Desktop Client](https://desktop.github.com/) or the [GitHub Command Line Interface](https://cli.github.com/).
+
+To use the CSV-to-CSV-W GitHub action,
+
+1. First fork our example [GitHub repository](https://github.com/GDonRanasinghe/csv-to-csvw). Select your GitHub username as the `Owner` and leave the `Repository name` as it is.
+![Fork Repository](../images/guides/csv-to-csvw-github-action/fork_repository.png)
+
+2. Then go to the repository settings and set the branch for GitHub pages - under the `Source` section, set the `Branch` to `gh-pages` and set the folder location to `/(root)`. Also, keep a note of the URL at which your GitHub Pages site is published at.
+![GitHub Pages Setting](../images/guides/csv-to-csvw-github-action/github_pages_setting.png)
+
+3. The repository already consists of example inputs (see the `csv/` folder) and the generated outputs (see the `out/` folder). The users can use these input examples to familiarise themselves with the criteria discussed in [Organising Inputs](#organising-inputs).
+
+4. Now that the repository is forked and the GitHub pages settings are configured, the inputs can be committed using the GitHub web console.
+![Commit Inputs](../images/guides/csv-to-csvw-github-action/commit_files.png)
+
+5. Once the inputs are committed, the action will start to run and will perform the steps discussed in [Key Steps Performed by the Action](#key-steps-performed-by-the-action). To see the progress of the action, go to the `Actions` section in the GitHub web console.
+![GitHub Action Running](../images/guides/csv-to-csvw-github-action/action_running.png)
+A more detailed view of the progress of the action can be seen by clicking on the action.
+![GitHub Action Running Detailed](../images/guides/csv-to-csvw-github-action/action_running_detailed.png)
+
+6. Once the CSV-to-CSV-W action is completed, another action called `pages build and deployment` will start to run. This action is responsible for deploying the outputs to the GitHub pages.
+![Pages Build and Deployment Action](../images/guides/csv-to-csvw-github-action/pages_build_action.png)
+
+7. Now we are ready to explore the outputs produced by the action. First look at the `out/` folder within the repository. If you are using the GitHub Desktop Client or the Github Command Line Interface, make sure to run `git pull` beforehand. The `out/` folder now consists of the CSV-Ws and inspect command logs generated for inputs committed to the repository.
```

Comment:
This section is better represented by an "output" section of the example. Additionally, an input section should be created to show people what you want to use to trigger the builds.